src/rdo.rs

```diff
@@ -291,9 +291,9 @@ pub fn rdo_mode_decision(
   let mode_context =
     cw.find_mvrefs(bo, LAST_FRAME, &mut mv_stack, bsize, false);
 
-  for &luma_mode in &mode_set {
+  mode_set.iter().for_each(|&luma_mode| {
```

Comment:
I'd like to learn what would be benefits by doing this way?

Reply:
It is an intermediate step to get to use iter_par and have rayon spawn a thread for each iteration.

Tomorrow I'll re-read the code since sleep-rebasing did not work that well :)

Reply:
Ideally from here we could go this way:

`for loop` -> `iter().for_each()` -> `iter().map().min_by()` -> `par_iter().map().min_by`.

Ideally evaluating each luma mode in parallel and then having a pass computing the best.

This would work better if we have the least amount of mutable state across and @barrbrain is already working on reducing it.

Using `flat_map()` so we could have each chroma evaluation going in parallel as well would require even more effort so I stopped thinking about it right now.